examples/cargo_without_cmake/build.rs

```diff
@@ -23,5 +29,43 @@ fn main() {
             println!("cargo:rerun-if-changed=src/cpp/run.cpp");
         })
         .build();
+
+    // This fails to link with GNU ld, which is the default on most Linux distributions, so use lld or mold instead.
+    #[cfg(unix)]
+    {
+        let flags = env::var("CARGO_ENCODED_RUSTFLAGS").unwrap();
+        // Don't override custom flags
+        if !flags.contains("-fuse-ld") {
+            // don't search for `GNU ld` because mold also includes that string in `mold --version` output
+            let linker_is_gnu = str::from_utf8(
+                &command_version_output("ld")
+                    .expect("Could not run ld command")
+                    .stdout,
+            )
+            .unwrap()
+            .contains(&"Free Software Foundation");
+
+            // linker is probably lld or mold already
+            if !linker_is_gnu {
+                return;
+            }
+
+            // mold is faster, but specifing mold with -fuse-ld requires GCC >= 12 or Clang.
+            // Unfortunately cargo does not provide a means to set the linker driver via build scripts,
+            // so linking would fail trying to use -fuse-ld=mold with GCC < 12 even if clang is installed.
+            // So, prefer lld to mold for robustness on the widest range of systems.
+            // mold can still be used by manually specifying it in ~/.cargo/config.toml or the RUSTFLAGS environment variable.
+            if command_version_output("lld").is_ok() {
+                println!("cargo:rustc-link-arg=-fuse-ld=lld");
+                return;
+            }
```

Comment:
any reason for not doing and if else if else ?

```
if lld {
...
} else if mold {
...
} else {
...
}
```

Reply:
Nope, that's simpler :)